### ISSUE TITLE
[BUG] column_values_match_regex metric fails with Trino

### ISSUE THREAD

**Issue by itaise** [great-expectations/great_expectations#8701]
Hi, 
We are trying to use "expect_column_values_to_match_regex" with Trino. 
The expectation is using column_values_match_regex metric - this metric calls get_dialect_regex_expression, on [this line](https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/expectations/metrics/util.py#L90). 

The call fails because the Trino SQLAlchemy dialect **has no attribute "dialect".** 
I think this part of redshift dialect should also have try/except like the others. 

Can you please check? 

 example: 
```
 {
            "expectation_type": "expect_column_values_to_match_regex",
            "kwargs":
            {
                "column": "thread_latest_topic_skill_name",
                "regex": "[^,]"
            }
        }

```
thanks a lot! 
GX version - 0.17.14

**Comment by itaise** [comment]
![image](https://github.com/great-expectations/great_expectations/assets/64745584/63e39b29-fb0d-4b28-9795-28a358707dd8)

**Comment by austiezr** [comment]
Hey @itaise ! Thanks for surfacing this. We'll review and be in touch; don't hesitate to reach out for support if you're interested in contributing here!

**Comment by itaise** [comment]
Thanks @austiezr ! My fix is in the attached PR. thanks!

**Comment by austiezr** [comment]
Resolved by #8707
